pkg/kv/kvclient/kvcoord/dist_sender.go

```diff
@@ -1688,9 +1688,19 @@ func maybeSwapErrorIndex(pErr *kvpb.Error, a, b int) {
 }
 
 // mergeErrors merges the two errors, combining their transaction state and
-// returning the error with the highest priority.
+// returning the error with the highest priority. If errors have the same
+// priority, the error with the lowest request index is preferred. This allows
+// a caller issuing multiple cputs to control which cput error is returned.
```

Comment:
Maybe instead of "cput error" it would be more accurate to say "ConditionFailedError" here.